src/ert/analysis/_es_update.py

```diff
@@ -153,56 +153,75 @@ def _get_observations_and_responses(
     observation_values = []
     observation_errors = []
     indexes = []
-    observations = ensemble.experiment.observations
-    for obs in selected_observations:
-        observation = observations[obs]
-        group = observation.attrs["response"]
-        all_responses = ensemble.load_responses(group, tuple(iens_active_index))
-        if "time" in observation.coords:
-            all_responses = all_responses.reindex(
-                time=observation.time,
-                method="nearest",
+    observations_by_type = ensemble.experiment.observations
+    for (
+        response_type,
+        response_cls,
+    ) in ensemble.experiment.response_configuration.items():
+        if response_type not in observations_by_type:
+            continue
+
+        observations_for_type = observations_by_type[response_type].filter(
+            polars.col("observation_key").is_in(list(selected_observations))
+        )
+        responses_for_type = ensemble.load_responses(
+            response_type, realizations=tuple(iens_active_index)
+        )
+
+        # Note that if there are duplicate entries for one
+        # response at one index, they are aggregated together
+        # with "mean" by default
+        pivoted = responses_for_type.pivot(
+            on="realization",
+            index=["response_key", *response_cls.primary_key],
+            aggregate_function="mean",
```

Comment:
What is the implication of `mean`?

Reply:
I said so in the comment 😅

Reply:
Will that be output somewhere? Is it possible to for example log it?

Reply:
It is for the edge case where we end up with duplicate values for one response at one index, for example a given time. In that case, we need to aggregate them for the pivoted table to make sense, else the `index` used to pivot contains duplicates. 

Example from running `test_that_duplicate_summary_time_steps_does_not_fail`:
```
responses_for_type.pivot(
            on="realization",
            index=["response_key", *response_cls.primary_key],
            aggregate_function="mean",
        )
Out[9]: 
shape: (1, 5)
┌──────────────┬─────────────────────┬───────────┬────────┬──────────┐
│ response_key ┆ time                ┆ 0         ┆ 1      ┆ 2        │
│ ---          ┆ ---                 ┆ ---       ┆ ---    ┆ ---      │
│ str          ┆ datetime[ms]        ┆ f32       ┆ f32    ┆ f32      │
╞══════════════╪═════════════════════╪═══════════╪════════╪══════════╡
│ FOPR         ┆ 2014-09-10 00:00:00 ┆ -1.603837 ┆ 0.0641 ┆ 0.740891 │
└──────────────┴─────────────────────┴───────────┴────────┴──────────┘
responses_for_type
Out[10]: 
shape: (4, 4)
┌─────────────┬──────────────┬─────────────────────┬───────────┐
│ realization ┆ response_key ┆ time                ┆ values    │
│ ---         ┆ ---          ┆ ---                 ┆ ---       │
│ u16         ┆ str          ┆ datetime[ms]        ┆ f32       │
╞═════════════╪══════════════╪═════════════════════╪═══════════╡
│ 0           ┆ FOPR         ┆ 2014-09-10 00:00:00 ┆ -1.603837 │
│ 1           ┆ FOPR         ┆ 2014-09-10 00:00:00 ┆ 0.0641    │
│ 2           ┆ FOPR         ┆ 2014-09-10 00:00:00 ┆ 0.740891  │
│ 2           ┆ FOPR         ┆ 2014-09-10 00:00:00 ┆ 0.740891  │
└─────────────┴──────────────┴─────────────────────┴───────────┘

```

Alternatively we could strive to achieve something like this:
```
┌──────────────┬─────────────────────┬───────────┬────────┬──────────┐
│ response_key ┆ time                ┆ 0         ┆ 1      ┆ 2        │
│ ---          ┆ ---                 ┆ ---       ┆ ---    ┆ ---      │
│ str          ┆ datetime[ms]        ┆ f32       ┆ f32    ┆ f32      │
╞══════════════╪═════════════════════╪═══════════╪════════╪══════════╡
│ FOPR         ┆ 2014-09-10 00:00:00 ┆ -1.603837 ┆ 0.0641 ┆ 0.740891 │
│ FOPR         ┆ 2014-09-10 00:00:00 ┆    NaN    ┆  NaN   ┆ 0.740891 │
└──────────────┴─────────────────────┴───────────┴────────┴──────────┘
```

Reply:
Could be logged / given as a warning somehow, I'm not so familiar with when/why it happens, which may be relevant to what the warning/logging message should be.

Reply:
(Performance-wise it might be slow to always check if some values were aggregated, or a naive try-catch around the pivot, as it will pass if there are no duplicate values)

Reply:
If there is a good, somewhat performant way of warning the user this has happened, that would be good. My hunch is that this would typically happen in pressure tests where the time resolution is quite high, and the simulator does not have the same resolution.

Reply:
Would it be OK to do this in a separate PR? I think the try-catch, first trying without an aggregation, then trying with one, should be easy to add / easy to remove if it turns out to have bad side effects. Should maybe be tested as its own thing just to be sure.